src/Views/VPNPage.vala

```diff
@@ -296,23 +301,28 @@ public class Network.VPNPage : Network.Widgets.Page {
         var selected_row = vpn_list.get_selected_row () as VPNMenuItem;
         if (selected_row != null && sel_row != null) {
             if (sel_row == selected_row) {
-                try {
-                    selected_row.connection.delete (null);
-                } catch (Error e) {
-                    warning (e.message);
-                    var dialog = new Granite.MessageDialog (
-                        _("Failed to remove VPN connection"),
-                        "",
-                        new ThemedIcon ("network-vpn"),
-                        Gtk.ButtonsType.CLOSE
-                    ) {
-                        badge_icon = new ThemedIcon ("dialog-error"),
-                        transient_for = (Gtk.Window) get_toplevel ()
-                    };
-                    dialog.show_error_details (e.message);
-                    dialog.run ();
-                    dialog.destroy ();
-                }
+                selected_row.connection.delete_async.begin (
+                    null,
+                    (obj, res) => {
+                        try {
+                            selected_row.connection.delete_async.end (res);
+                        } catch (Error e) {
+                            warning (e.message);
+                            var dialog = new Granite.MessageDialog (
+                                _("Failed to remove VPN connection"),
+                                "",
+                                new ThemedIcon ("network-vpn"),
+                                Gtk.ButtonsType.CLOSE
+                            ) {
+                                    badge_icon = new ThemedIcon ("dialog-error"),
+                                        transient_for = (Gtk.Window) get_toplevel ()
+                                        };
```

Comment:
Indentation seems to have gotten a little messed up here :sweat_smile:

Reply:
🤦